src/Microsoft.Tye.Core/KubernetesManifestGenerator.cs

```diff
@@ -274,6 +274,36 @@ internal static class KubernetesManifestGenerator
             spec = new YamlMappingNode();
             template.Add("spec", spec);
 
+            if (project.Sidecars.Count > 0)
+            {
+                // Share process namespace when we have sidecars. So we can list other processes.
+                // see: https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/#understanding-process-namespace-sharing
+                spec.Add("shareProcessNamespace", new YamlScalarNode("true") { Style = ScalarStyle.Plain });
+            }
+
+            if (project.RelocateDiagnosticsDomainSockets)
+            {
+                // Our diagnostics functionality uses $TMPDIR to locate other dotnet processes through
+                // eventpipe. see: https://github.com/dotnet/diagnostics/blob/master/documentation/design-docs/ipc-protocol.md#transport
+                //
+                // In order for diagnostics features to 'find' each other, we need to make $TMPDIR into
+                // something shared.
+                //
+                // see: https://kubernetes.io/docs/tasks/access-application-cluster/communicate-containers-same-pod-shared-volume/
+                project.EnvironmentVariables.Add(new EnvironmentVariableBuilder("TMPDIR")
+                {
+                    Value = "/var/tye/diagnostics",
+                });
+
+                foreach (var sidecar in project.Sidecars)
+                {
+                    sidecar.EnvironmentVariables.Add(new EnvironmentVariableBuilder("TMPDIR")
+                    {
+                        Value = "/var/tye/diagnostics",
+                    });
+                }
+            }
+
```

Comment:
All of these pieces took a while to figure out. Basically there's a few things in the way if you want to use eventpipe in a sidecar:

- You need eventpipe to find the other process
- You need a way to find out which process you want (alternatively handle all of them)
  - event pipe just gives you PIDs and that's all
  - I took the approach of using `ps -p <pid> -o command=` to output the args of each process
- If you share process space (needed to enumerate processes in another container) then the target process won't be pid=1.

So basically it gets much more complicated once the number of processes is > 2 (you + one other .net process). I wanted to be resilient for the case where someone did a `dotnet run` or something like that.

Open to ideas about how to make this better.